File: dotnet/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example2/Models/Order.cs
namespace MediatorDesignPattern.Example2.Models;

public class Order
{
    public Guid Id { get; set; }
    public string CustomerName { get; set; } = string.Empty;
    public string ProductName { get; set; } = string.Empty;
    public int Quantity { get; set; }
    public decimal Price { get; set; }
    public decimal TotalAmount => Quantity * Price;
    public OrderStatus Status { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
}

public enum OrderStatus
{
    Pending,
    Confirmed,
    Shipped,
    Delivered,
    Cancelled
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example2/src/Models/Order.ts
export enum OrderStatus {
  Pending = 'Pending',
  Confirmed = 'Confirmed',
  Shipped = 'Shipped',
  Delivered = 'Delivered',
  Cancelled = 'Cancelled',
}

export class Order {
  id: string;
  customerName: string;
  productName: string;
  quantity: number;
  price: number;
  get totalAmount(): number { return this.quantity * this.price; }
  status: OrderStatus;
  createdAt: Date;
  updatedAt?: Date;

  constructor(data: Omit<Order, 'totalAmount'>) {
    this.id = data.id;
    this.customerName = data.customerName;
    this.productName = data.productName;
    this.quantity = data.quantity;
    this.price = data.price;
    this.status = data.status;
    this.createdAt = data.createdAt;
    this.updatedAt = data.updatedAt;
  }
}
